pycbc/waveform/generator.py

```diff
@@ -517,6 +520,10 @@ def generate(self, **kwargs):
                 hp = apply_fd_time_shift(hp, self.current_params['tc']+tshift,
                                          copy=False)
             h['RF'] = hp
+        if self.gates is not None:
+            # resize all to nearest power of 2
+            [d.resize(ceilpow2(len(d)-1)+1) for d in h.values()]
```

Comment:
This is a style thing but this look really awkward. Its making a list that doesn't get assigned. It might make more sense to do:
```
for d in h.values():
    d.resize(ceilpow2(len(d)-1)+1)
```
Is there any reason not to?

Reply:
Fixed. I was thinking of the "list comprehension if faster than for loop" maxim, but you're right, that's only true if you are creating a list.